centreon/src/Centreon/Infrastructure/MonitoringServer/MonitoringServerRepositoryRDB.php

```diff
@@ -269,6 +270,65 @@ public function findServer(int $monitoringServerId): ?MonitoringServer
         return null;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findServerByIdAndAccessGroups(int $monitoringServerId, array $accessGroups): ?MonitoringServer
+    {
+        if ($accessGroups === []) {
+            return null;
+        }
+
+        $accessGroupIds = array_map(
+            fn($accessGroup) => $accessGroup->getId(),
+            $accessGroups
+        );
+
+        if (! $this->hasRestrictedAccessToMonitoringServers($accessGroupIds)) {
+            return $this->findServer($monitoringServerId);
+        }
```

Comment:
Useless, if you've reached this point, it's because you have restrictions.

Reply:
In ACL resources `acl_resources_poller_relation` is a filter, if there're no pollers in filter that means the user have access to all pollers. That's why we need to check it before.